src/java.base/share/classes/java/io/Console.java

```diff
@@ -619,31 +402,17 @@ private static Console instantiateConsole(boolean istty) {
                         .filter(Objects::nonNull)
                         .findAny()
                         .map(jc -> (Console) new ProxyingConsole(jc))
-                        .orElse(istty ? new Console() : null);
+                        .orElse(istty ? new ConsoleImpl() : null);
             };
             return AccessController.doPrivileged(pa);
         } catch (ServiceConfigurationError ignore) {
             // default to built-in Console
-            return istty ? new Console() : null;
+            return istty ? new ConsoleImpl() : null;
         }
     }
 
     private static final Console cons;
     private static native boolean istty();
 
-    Console() {
-        readLock = new Object();
-        writeLock = new Object();
-        out = StreamEncoder.forOutputStreamWriter(
-                  new FileOutputStream(FileDescriptor.out),
-                  writeLock,
-                  CHARSET);
-        pw = new PrintWriter(out, true) { public void close() {} };
-        formatter = new Formatter(out);
-        reader = new LineReader(StreamDecoder.forInputStreamReader(
-                     new FileInputStream(FileDescriptor.in),
-                     readLock,
-                     CHARSET));
-        rcb = new char[1024];
-    }
+    Console() {}
```

Comment:
What would you think about moving the no-arg constructor to the top of the method as it's not easy to spot.